flume-ng-sinks/flume-hdfs-sink/src/main/java/org/apache/flume/sink/hdfs/HDFSEventSink.java

```diff
@@ -415,6 +410,9 @@ public void run(String bucketPath) {
           }
           bucketWriter.append(event);
         }
+
+        // track the buckets getting written in this transaction
+        writers.add(bucketWriter);
```

Comment:
Looks good to also capture the bucketwriter created on line 406. Since the original (line 389 or 393) one threw a closed exception that means that we don't have to flush the original one only the newly created one (close implicitly contains flush as well)